spras/btb.py

```diff
@@ -0,0 +1,188 @@
+        if not sources or not targets or not edges or not output_file:
+            raise ValueError('Required BowtieBuilder arguments are missing')
+
+        # Test for pytest (docker container also runs this)
+        # Testing out here avoids the trouble that container errors provide
+        if not Path(sources).exists() or not Path(targets).exists() or not Path(edges).exists():
+            raise ValueError('Missing input file')
+
+        # Testing for btb index
+        # It's a bit messy, but it works \_('_')_/
+        with open(edges, 'r') as edge_file:
+            try:
+                for line in edge_file:
+                    line = line.strip()
+                    line = line.split('\t')
+                    line = line[2]
+                    
+            except Exception as err:
+                print("error!!")
+                print(err)
+                raise(err)
+
+        work_dir = '/btb'
+
+        # Each volume is a tuple (src, dest)
+        volumes = list()
+
+        bind_path, source_file = prepare_volume(sources, work_dir)
+        volumes.append(bind_path)
+
+        bind_path, target_file = prepare_volume(targets, work_dir)
+        volumes.append(bind_path)
+
+        bind_path, edges_file = prepare_volume(edges, work_dir)
+        volumes.append(bind_path)
+
+        # PathLinker does not provide an argument to set the output directory
+        # Use its --output argument to set the output file prefix to specify an absolute path and prefix
+        out_dir = Path(output_file).parent
+        # PathLinker requires that the output directory exist
+        out_dir.mkdir(parents=True, exist_ok=True)
+        bind_path, mapped_out_dir = prepare_volume(str(out_dir), work_dir)
+        volumes.append(bind_path)
+        mapped_out_prefix = mapped_out_dir + '/raw-pathway.txt'  # Use posix path inside the container
+
+        command = ['python',
+                   'btb.py',
+                   '--edges',
+                   edges_file,
+                   '--sources',
+                   source_file,
+                   '--target',
+                   target_file,
+                   '--output',
+                   mapped_out_prefix]
+        # command = ['ls', '-R']
+
+
+        print('Running BowtieBuilder with arguments: {}'.format(' '.join(command)), flush=True)
+
+        container_suffix = "bowtiebuilder"
+        out = run_container(container_framework,
+                            container_suffix,
+                            command,
+                            volumes,
+                            work_dir)
+        print(out)
+        print("Source file: ", source_file)
+        print("target file: ", target_file)
+        print("edges file: ", edges_file)
+        print("mapped out dir: ", mapped_out_dir)
+        print("mapped out prefix: ", mapped_out_prefix)
+
+
+
+
+
+        # Rename the primary output file to match the desired output filename
```

Comment:
I think we can condense these comments to say "output is already written to raw-pathway.txt file."